### PR TITLE
Error pages

### PR DESCRIPTION
**Pull request by phil303**
They look like this:

screen shot 2016-07-21 at 10 04 46 am

### CODE REVIEW

**Comment by phil303** [comment]
:eyeglasses: @schwers or @nramadas

**Comment by nramadas** [comment]
so, this takes care of bad urls within the normal client side flow, right? what happens if the user hits the node server with a bad url?

File: src/app/components/ErrorPage/index.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const GO_BACK = 'Go Back!';
+const GO_TO_FRONTPAGE = 'To the front page!';
+
+export default (props) => {
+  // NOTE: the second condition is a side-effect of a bug in our platform
+  // code, which is setting the url's referrer to itself. Ideally, only the
+  // first condition is needed.
+  const toFrontpage = props.referrer === '' || props.referrer === props.url;
+
+  const href = toFrontpage ? props.referrer : '/';
```
Comment (nramadas) [inline src/app/components/ErrorPage/index.jsx]: shouldn't this be `toFrontpage ? '/' : props.referrer`?

Comment (phil303) [inline src/app/components/ErrorPage/index.jsx]: yes! good find. I inverted the logic on the boolean (used to be called `goBack`) and I forgot to update this.

**Comment by schwers** [comment]
Just as note status codes aren't mutually exclusive from the html ;)

Should this be renamed 404Page versus ErrorPage?

**Comment by nramadas** [comment]
lgtm :fish:

**Comment by phil303** [comment]
@schwers that thought had crossed my mind and originally I thought this might handle other page errors but I'm actually not sure what those would be since resources are fetched with ajax. I'd be fine with renaming.

**Comment by phil303** [comment]
@nramadas @schwers regarding returning a 404. That's an excellent point that I completely forgot to address. Is there a way to return 404 status codes with platform?

**Comment by nramadas** [comment]
Well, the other error pages would be when the server 500s, right?

**Comment by phil303** [comment]
:haircut:

**Comment by schwers** [comment]
Should `/handlers/Status404PageHandler.js` be `/handlers/Status404Page.js` with our new naming conventions?

**Comment by phil303** [comment]
@schwers yes. Good catch.

**Comment by schwers** [comment]
🐠  part two, the 🍣

**Pull request merged**